src/components/application_manager/src/resumption/resumption_data_json.cc

```diff
@@ -238,11 +254,13 @@ bool ResumptionDataJson::RemoveApplicationFromSaved(
     const std::string& policy_app_id, const std::string& device_id) {
   using namespace app_mngr;
   LOG4CXX_AUTO_TRACE(logger_);
-  sync_primitives::AutoLock autolock(resumption_lock_);
+
   bool result = false;
   std::vector<Json::Value> temp;
-  for (Json::Value::iterator it = GetSavedApplications().begin();
-       it != GetSavedApplications().end();
+  resumption::LastStateAccessor accessor = last_state_wrapper_->get_accessor();
+  Json::Value dictionary = accessor.GetData().dictionary();
+  Json::Value& saved_apps = GetSavedApplications(dictionary);
+  for (Json::Value::iterator it = saved_apps.begin(); it != saved_apps.end();
```

Comment:
@YarikMamykin same as above

Reply:
@AByzhynar fixed in https://github.com/smartdevicelink/sdl_core/pull/2626/commits/723596386f8f684930582be36f8ea2d1659bd7c1